### PR TITLE
Update riello (ser, usb) for shutdown/reboot delays (Fixes #530)

### LINKED ISSUE #530: Riello IDG400 wrong values & not powering off
Hello,
I just changed my old UPS with a new Riello IDG400:
1: it shows battery charge 255% - battery.runtime 3932100 - Runtime (H:M:S): 1092:15:00
is this an hardware problem or something on Nut side?
2: I use a raspberry as nut server to shutdown all lan clients after 180 seconds without main power; the old ups was also powered off after FINALDELAY 5 seconds while the Riello keeps staying on battery; is this something related to the first problem?
Thanks

### PR DESCRIPTION
**Pull request by geoghegan**
This should address the issues in #530. Tested with an IDG1600

Output from `upsc` below

```
battery.capacity: 9
battery.voltage: 27.0
battery.voltage.nominal: 24
device.mfr: RPS S.p.a.
device.model: UNV1
device.serial:
device.type: ups
driver.name: riello_usb
driver.parameter.pollinterval: 2
driver.parameter.port: /dev/ttyUSB0
driver.parameter.synchronous: no
driver.version: 2.7.4-4547-gc911e825
driver.version.internal: 0.07
input.bypass.frequency: 409.50
input.bypass.voltage: 4095
input.frequency: 50.10
input.voltage: 240
output.frequency: 50.10
output.frequency.nominal: 50.0
output.L1.current: 0
output.L1.power: 0
output.L1.realpower: 0
output.L2.current: 0
output.L2.power: 0
output.L2.realpower: 0
output.L3.current: 0
output.L3.power: 0
output.L3.realpower: 0
output.power.percent: 0
output.voltage: 237
output.voltage.nominal: 230
ups.delay.reboot: 5
ups.delay.shutdown: 5
ups.firmware: SWM038-02-04
ups.load: 0
ups.mfr: RPS S.p.a.
ups.model: UNV1
ups.power.nominal: 1600
ups.productid: 5500
ups.realpower.nominal: 960
ups.serial:
ups.status: OL
ups.vendorid: 04b4
```

### CODE REVIEW

**Comment by jimklimov** [comment]
Thanks, sorry for having missed that comment in https://github.com/networkupstools/nut/issues/530#issuecomment-1015560788

Compared to the patch quoted there (from https://www.mail-archive.com/nut-upsdev@alioth-lists.debian.net/msg00106.html message) this PR misses the `riello_ser.c` fix to be on par. Would you please paste that one in also?

Additionally, the dstate assignment from `unsigned int` should not be formatted as `%d` but as `%u`, right?

File: drivers/riello_usb.c [hunk 1]
```diff
@@ -972,6 +979,13 @@ void upsdrv_initinfo(void)
 	dstate_addcmd("test.battery.start");
 	dstate_addcmd("test.panel.start");
 
+	dstate_setinfo("ups.delay.shutdown", "%d", offdelay);
+	dstate_setflags("ups.delay.shutdown", ST_FLAG_RW | ST_FLAG_STRING);
+	dstate_setaux("ups.delay.shutdown", 3);
+	dstate_setinfo("ups.delay.reboot", "%d", bootdelay);
```
Comment (jimklimov) [inline drivers/riello_usb.c:985]: Here and above, the dstate assignment from `unsigned int` should not be formatted as `%d` but as `%u`, right?

**Comment by geoghegan** [comment]
It _should_ address #957 as `ups.delay.shutdown` is no longer undefined / NULL in this PR.

**Comment by geoghegan** [comment]
I had a look through the open issues - Looks like this PR may address #530 , #957, and at least a portion of #738

**Comment by geoghegan** [comment]
How do I set variables on the UPS itself, @jimklimov ?

e.g. if I want to set `ups.delay.shutdown` and `ups.delay.reboot`? My concern is the static definition of 5 seconds in this PR. I want to check that a user assigned override will actually pass through to the UPS.

I have attempted the following

```
./drivers/riello_usb -a ups -x ups.delay.reboot=10
Network UPS Tools - Riello USB driver 0.07 (2.7.4-4549-ga1572032)
Warning: This is an experimental driver.
Some features may not function correctly.


Fatal error: 'ups.delay.reboot' is not a valid variable name for this driver.```

**Comment by jimklimov** [comment]
I think the `override.*` namespace for arguments should help here, e.g. `override.ups.delay.reboot = 10`

**Comment by geoghegan** [comment]
Yep! I was able to set the override, and `upsc` shows the correct value when polling the UPS.

Thanks for the merge :-)

**Comment by NilByte** [comment]
@geoghegan Thanks for your effort. Appreciate it a lot. :)

I build NUT according to:

https://github.com/networkupstools/nut/wiki/Building-NUT-on-Debian,-Raspbian-and-Ubuntu

Unsupported parameters are gone as expected and new parameters ups.delay.reboot and ups.delay.shutdown now listed as expected.

`upscmd -u admin -p ups ups shutdown.return` is giving result OK

I can set new values for ups.delay.reboot and ups.delay.shutdown but as you experience, these don't survive a reboot. What worked for me as example was:

`sudo /lib/nut/riello_usb -a <ups> -x override.ups.delay.reboot=10`

Testing power failure however, doesn't end up cutting the power to the UPS as I had hoped it would, so at least for me it seems like the part on not powering off mentioned in https://github.com/networkupstools/nut/issues/530 was not solved by this update.

Have you tested power failure and was the UPS finally powering off?

I also expected that it would now be possible to use offdelay and bootdelay in ups.conf to override defaults values of ups.delay.shutdown and ups.delay.reboot with this driver version, but as far as I experience setting these are not working.

upsc result showing driver version

```
Init SSL without certificate database
battery.capacity: 7
battery.voltage: 13.5
battery.voltage.nominal: 12
device.mfr: RPS S.p.a.
device.model: UNV1
device.serial:
device.type: ups
driver.name: riello_usb
driver.parameter.pollinterval: 2
driver.parameter.port: auto
driver.parameter.synchronous: no
driver.version: 2.7.4-4609-g61862059
driver.version.internal: 0.07
input.bypass.frequency: 409.50
input.bypass.voltage: 4095
input.frequency: 50.00
input.voltage: 230
output.frequency: 50.00
output.frequency.nominal: 50.0
output.L1.current: 0
output.L1.power: 0
output.L1.realpower: 0
output.L2.current: 0
output.L2.power: 0
output.L2.realpower: 0
output.L3.current: 0
output.L3.power: 0
output.L3.realpower: 0
output.power.percent: 0
output.voltage: 226
output.voltage.nominal: 230
ups.delay.reboot: 10
ups.delay.shutdown: 5
ups.firmware: SWM038-01-02
ups.load: 0
ups.mfr: RPS S.p.a.
ups.model: UNV1
ups.power.nominal: 400
ups.productid: 5500
ups.realpower.nominal: 240
ups.serial:
ups.status: OL
ups.vendorid: 04b4
```

**Comment by NilByte** [comment]
#530 , #976 and #750 could probably also be closed now.

**Pull request merged**